workflows/mcmicro.nf

```diff
@@ -76,12 +76,55 @@ workflow MCMICRO {
     // // Run Background Correction
     // BACKSUB(ASHLAR.out.tif, ch_markers)
     //BACKSUB(ASHLAR.out.tif, [[id: "backsub"], params.marker_sheet])
+    /*
+    if (params.backsub) {
+        BACKSUB(ASHLAR.out.tif, [[id:"$ASHLAR.out.tif[0]['id']"], params.marker_sheet])
+        ch_segmentation_input = BACKSUB.out.backsub_tif
+        ch_versions = ch_versions.mix(BACKSUB.out.versions)
+    } else {
+    */
+        ch_segmentation_input = ASHLAR.out.tif
+    /*
+    }
+    */
+
     //ch_versions = ch_versions.mix(BACKSUB.out.versions)
 
     // Run Segmentation
 
-    DEEPCELL_MESMER(ASHLAR.out.tif, [[:],[]])
-    ch_versions = ch_versions.mix(DEEPCELL_MESMER.out.versions)
+    params.segmentation_list = params.segmentation.split(',') as List
+    mesmer_out_mask = channel.empty()
+    if ("mesmer" in params.segmentation_list) {
+        DEEPCELL_MESMER(ch_segmentation_input, [[:],[]])
+        ch_versions = ch_versions.mix(DEEPCELL_MESMER.out.versions)
+        ch_segmentation_input
```

Comment:
What's this section doing? Is it only to add `segmenter: 'mesmer'` to the meta on the output? If so, it would probably be simpler to add that to the _input_ meta and use the meta pass-through from the module. Something like this:
```groovy
ASHLAR.out.tif
    .multiMap{ meta, image ->
        img: [meta + segmenter: 'mesmer', image]
        membrane_img: [[:], []]
    }
    | DEEPCELL_MESMER
```

Reply:
I thought I needed those conditions, but I've fixed that in a previous commit.
I forgot I could name those variable, so I updated all the muliMaps. Thanks for the reminder!
I had to add another set of brackets to get the append to meta to work, but it seems to be working now.

Reply:
I think it could still be simpler. Instead of taking the output of each segmenter module and doing a join and setting the result into a new variable, you can modify the _input_ into those modules with the necessary extra meta keys and that meta will pass through to the output. This also lets us preserve the simple `foo.bar... | PROCESS` style that we've been using so far (which I think reads better). I can write out a more fully fleshed-out example but I haven't tested it:
```nextflow
ch_masks = Channel.empty()

ch_segmentation_input
    .multiMap{ meta, image ->
        img: [meta + [segmenter: 'mesmer'], image]
        membrane_img: [[:], []]
    }
    | DEEPCELL_MESMER
ch_masks = ch_masks.mix(DEEPCELL_MESMER.out.mask)
ch_versions = ch_versions.mix(DEEPCELL_MESMER.out.versions)

ch_segmentation_input
    .multiMap{ meta, image ->
        image: [meta + [segmenter: 'cellpose'], image]
        model: params.cellpose_model
    }
    | CELLPOSE
ch_masks = ch_masks.mix(CELLPOSE.out.mask)
ch_versions = ch_versions.mix(CELLPOSE.out.versions)
```

Reply:
Oh, I see.  Yes, this is much cleaner.  I just tested it and it passes all tests with no changes.